climakitae/uncertain_utils.py

```diff
@@ -226,9 +229,25 @@ def _drop_member_id(dset_dict):
             dset_dict.update({dname: dset})  # Update dataset in dictionary
     return dset_dict
 
+def _precip_flux_to_total(ds):
+    """
+    converts precip flux units 
+    (kg m-2 s-1) to total precip
+    per month (mm)
+    NOTE: assumes regular calendar
+    """
+    ds_attrs = ds.attrs
+    days_month = ds.time.dt.days_in_month
+    seconds_month = 86400*days_month
+    ds = ds*seconds_month
+    ds = xr.where(ds>0.1,ds,np.nan)
+    ds.attrs = ds_attrs
+    ds.pr.attrs["units"] = 'mm'
+    return ds
```

Comment:
I think this could be simplified to avoid having to reset all the attributes: 
```
def _precip_flux_to_total(ds):
    """
    converts precip flux units 
    (kg m-2 s-1) to total precip
    per month (mm)
    NOTE: assumes regular calendar
    """
    days_month = ds.time.dt.days_in_month
    seconds_month = 86400*days_month
    ds = ds*seconds_month
    ds = ds.where(ds>0.1,np.nan)
    ds.pr.attrs["units"] = 'mm'
    return ds
```

Reply:
in my experience it did lose attributes when I didn't copy them over manually, even if I used xarray's built-in `xarray.set_options(keep_attrs=True`